Example/Tests/ListRuleIntegrationTests.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  Created by Maren Osnabrug on 27-05-16.
+//  Copyright © 2016 CocoaPods. All rights reserved.
+//
+
+import XCTest
+@testable import MarkyMark
+
+class ListRuleIntegrationTests: XCTestCase {
+
+    var sut:ListRule!
+
+    override func setUp() {
+        super.setUp()
+        sut = ListRule(listTypes: [AlphabeticListType(), OrderedListType(), UnOrderedListType()])
+    }
+
+    func testRecognizesLines() {
+        // Arrange
+        let markDownString = "- Hello"
+            + "- Hello"
+            + "  1. Hello"
+            + "  2. Hello"
+            + "    a. Hello"
+            + "    b. Hello"
+            + "    c. Hello"
+            + "      - Hello"
+            + "- Hello"
+
+        //Assert
+        XCTAssert(sut.recognizesLines(["- List item"]))
+        XCTAssert(sut.recognizesLines(["1. List item"]))
+        XCTAssert(sut.recognizesLines(["a. List item"]))
+        XCTAssert(sut.recognizesLines(["- List item", "1. List item", "a. List item"]))
+        XCTAssert(sut.recognizesLines([markDownString]))
+    }
+
+    func testCreateMarkDownItemWithLinesCreatesCorrectItem() {
+        //Act
+        let markdownItem = sut.createMarkDownItemWithLines(["- List item"])
+        let markdownItem2 = sut.createMarkDownItemWithLines(["4. List item"])
+        let markdownItem3 = sut.createMarkDownItemWithLines(["g. List item"])
+        let combinedMarkDownItems = sut.createMarkDownItemWithLines(["- List item","- List item","5. Numbered item"])
+        let combinedMarkDownItems2 = sut.createMarkDownItemWithLines(["A. Ordered item","4. Ordered item","- List item"])
+        let combinedMarkDownItems3 = sut.createMarkDownItemWithLines(["9. Numbered item","- List item","C. Ordered item"])
+
+        //Assert
+        XCTAssert(markdownItem is ListMarkDownItem)
+        XCTAssert(markdownItem2 is OrderedListMarkDownItem)
+        XCTAssert(markdownItem3 is AlphabeticallyOrderedMarkDownItem)
+        XCTAssert(combinedMarkDownItems is UnorderedListMarkDownItem)
+        XCTAssert(combinedMarkDownItems2 is AlphabeticallyOrderedMarkDownItem)
+        XCTAssert(combinedMarkDownItems3 is OrderedListMarkDownItem)
+    }
+
+    func testCreateMarkDownItemWithLinesContainsCorrectText() {
+        //Act
```

Comment:
I think we should use different texts for each check, so we also test it can be something different than "List Item"